aea/protocols/base.py

```diff
@@ -88,8 +86,8 @@ def sender(self) -> Address:
 
         :return the address
         """
-        assert self._sender is not None, "Sender must not be None."
-        return self._sender
+        assert self.has_sender, "Message's 'Sender' field must be set."
+        return self._sender  # type: ignore
```

Comment:
Role back the changes to avoid using type ignore...

Reply:
Done